### PR TITLE
INTYGFV-16560: Added aggregator for renew

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

File: web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        if (certificateToRenew == null) {
+            throw new IllegalStateException(
+                String.format("Certificate service returned null when getting certificate '%s'", certificateId)
+            );
+        }
+
+        final var integrationParameters = webCertUserService.getUser().getParameters();
+        final var renewalCertificate = csIntegrationService.renewCertificate(
+            certificateId,
+            csIntegrationRequestFactory.renewCertificateRequest(
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java:69]: Should this be part of the csIntegrationRequestFactory.renewCertificateRequest instead? What I mean is that the "alternateSsn" and "reference" code could be part of the request factory.

The logic needs to be managed for multiple type of requests, today we know that it is renew and replace.

File: web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                isAlternateSSNDefined(integrationParameters) ? integrationParameters.getAlternateSsn()
+                    : certificateToRenew.getMetadata().getPatient().getPersonId().getId(),
+                integrationParameters != null ? integrationParameters.getReference() : null
+            )
+        );
+
+        if (renewalCertificate == null) {
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java:76]: I've been thinking about this a few times before... it is really relevant to do null check on the return-value? Shouldn't you be able to trust the renewCertificate to always return a value and if something is wrong always throw an exception?

File: web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            throw new IllegalStateException(
+                String.format("Certificate service returned null when getting certificate '%s'", certificateId)
+            );
+        }
+
+        final var integrationParameters = webCertUserService.getUser().getParameters();
+        final var renewalCertificate = csIntegrationService.renewCertificate(
+            certificateId,
+            csIntegrationRequestFactory.renewCertificateRequest(
+                isAlternateSSNDefined(integrationParameters) ? integrationParameters.getAlternateSsn()
```
Comment (mww-sopra-steria) [inline web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java:70]: I found some code in the current facade-implementation that manages a situation if the personId is a "reserve id". See line 71 in RenewCertificateFacadeServiceImpl. Good to look at the whole commit that includes changes in multiple classes.

**Review by mww-sopra-steria — CHANGES_REQUESTED** [review]
Looks good! Have a few refactoring suggestion. Also I noticed some special handling of reserve id in current Webcert-code that we should follow up on.

File: web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java (hunk 2 above)
Comment (mww-sopra-steria) [reply web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java:68]: How about this question? :-)

File: web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java (hunk 3 above)
Comment (mww-sopra-steria) [reply web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java]: No, I don't think it is necessary. Any reserveIds handled by Webcert due to IntegrationParameters should stay in Webcert.

File: web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java (hunk 2 above)
Comment (mhornfeldt) [reply web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java]: I had two so I removed the first one only first. Now both are gone.

**Review by mww-sopra-steria — APPROVED** [review]
Looks good!

**Pull request merged**